### PR TITLE
Generate solution files automatically

### PR DESCRIPTION
**Pull request by jskeet**
This will:

- Create new solution files where they're missing
- Add all project references to solution files

It will not:

- Remove unused projects

### CODE REVIEW

File: tools/Google.Cloud.Tools.ProjectGenerator/Program.cs [hunk 1]
```diff
@@ -79,10 +84,68 @@ static void GenerateProjects(string apiRoot, ApiMetadata api)
...
+                WorkingDirectory = root,
+                RedirectStandardOutput = true
+            };
+            var process = Process.Start(psi);
+            // We assume there isn't so much output that this will block. Otherwise we'd have to read it in a different thread etc.
+            // 10s limit stops us from hanging forever...
+            process.WaitForExit(10000);
+            if (process.ExitCode != 0)
+            {
+                var text = process.StandardOutput.ReadToEnd();
```
Comment (chrisdunelm) [inline tools/Google.Cloud.Tools.ProjectGenerator/Program.cs]: StandardError may well contain the actual error on failure.

Comment (jskeet) [inline tools/Google.Cloud.Tools.ProjectGenerator/Program.cs]: Good point. Now capturing and reporting both.

**Comment by jskeet** [comment]
Will merge when I can get AppVeyor to go green.

**Comment by jskeet** [comment]
AppVeyor appears to be broken at the moment. Merging this while I investigate.

**Pull request merged**